composer/datasets/in_context_learning_evaluation.py

```diff
@@ -926,227 +1007,152 @@ class InContextLearningCodeEvalDataset(Dataset):
         destination_path (str): Temporary path to store downloaded datasets
         code_prelimiter (str): String to put before each code prompt (e.g. 'Q: ')
         fewshot_random_seed (int): Random seed to use for fewshot sampling
-        generations_per_sample: how many outputs to generate per prompt
-        top_p: top_p sampling parameter for nucleus sampling
-        top_k: top_k sampling parameter for number of samples to consider
+        generations_per_sample (int): how many outputs to generate per prompt
+        # TODO: is this correct?
```

Comment:
which?

Reply:
generations_per_sample mostly, but pass_at_k, top_p, and top_k as well would be nice.

Additionally, top_p and top_k are technically setable here but they are never declared or passed in in `build_icl_dataloader`, so they'll always be the default. Do we want them to be setable parameters?

If so, would we prefer to have an argument like "generator_kwargs" be added onto the batches' `generation_kwargs` instead of adding a specific few parameters like temperature? This would be a more general approach and extend to all tasks, not just CodeEval. More general is usually less complex, but it would likely be breaking given that `get_icl_task_dataloader` would probably need to change.

Reply:
updated to use generation_kwargs